Content.Server/Interaction/InteractionPopupSystem.cs

```diff
@@ -0,0 +1,86 @@
+using Content.Server.Popups;
+using Content.Server.Interaction.Components;
+using Content.Shared.Audio;
+using Content.Shared.Interaction;
+using Content.Shared.MobState;
+using Robust.Shared.Audio;
+using Robust.Shared.GameObjects;
+using Robust.Shared.IoC;
+using Robust.Shared.Localization;
+using Robust.Shared.Player;
+using Robust.Shared.Timing;
+using Robust.Shared.Random;
+
+
+namespace Content.Server.Interaction;
+
+public sealed class InteractionPopupSystem : EntitySystem
+{
+    [Dependency] private readonly IGameTiming _gameTiming = default!;
+    [Dependency] private readonly PopupSystem _popupSystem = default!;
+    [Dependency] private readonly IRobustRandom _random = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<InteractionPopupComponent, InteractHandEvent>(OnInteractHand);
+        SubscribeLocalEvent<InteractionPopupComponent, MobStateChangedEvent>(OnMobStateChanged);
+    }
+
+    private void OnMobStateChanged(EntityUid uid, InteractionPopupComponent component, MobStateChangedEvent args)
+    {
+        var originalSuccessChance = component.SuccessChance; // Stores this for later in case mob state changes back and forth.
+
+        if (!args.Component.IsAlive()) // if the mob being interacted with is not alive (dead, incapacitated, etc.)
+        {
+            component.SuccessChance = -1.0f; // set to the "invalid" value, which suppresses all popups and sound effects.
+            return;
+        }
+
+        if (args.Component.IsAlive()) // if the mob being interacted with is alive and well
+        {
+            component.SuccessChance = originalSuccessChance;
+            return;
+        }
+    }
+
+    private void OnInteractHand(EntityUid uid, InteractionPopupComponent component, InteractHandEvent args)
+    {
+        if (args.Handled)
+            return;
+
+        if (component.InteractDelay.TotalSeconds <= 0)
+            return;
```

Comment:
whats the impetus behind checking for this?

Reply:
True, I guess there's not much reason to if we've initialized InteractDelay correctly. I ported this from WindowSystem (Server) which handles knocking on windows, but this may be oldcode. Should I remove?

Reply:
Went ahead and removed it.